src/gui/addnewtorrentdialog.cpp

```diff
@@ -362,14 +391,16 @@ void AddNewTorrentDialog::saveSavePathHistory() const
     QDir selectedSavePath(ui->savePath->selectedPath());
     // Get current history
     QStringList history = settings()->loadValue(KEY_SAVEPATHHISTORY).toStringList();
+    if (history.size() > savePathHistoryLength())
+        history = history.mid(0, savePathHistoryLength());
```

Comment:
This list modification will not be persisted, when `if (!historyDirs.contains(selectedSavePath)) {`, because `settings()->storeValue(KEY_SAVEPATHHISTORY, history);` is called in that `if` block.

Reply:
Do you suggest to save it immediately after the optional trimming?

Reply:
Hmm, I don't know how you resolve this, I would move the history trimming at the end of the method, add some flag `bool historyChanged` and on the base of this flag call `settings()->storeValue()` as last thing in this method, something like this:
```
    foreach (const QString dir, history)
        historyDirs << QDir(dir);
    if (!historyDirs.contains(selectedSavePath)) {
        // Add save path to history
        history.push_front(selectedSavePath.absolutePath());
        // Limit list size
        if (history.size() > savePathHistoryLength()) {
            history.pop_back();
            historyChanged = true;
        }
    }
    // Used, when Save Path length changed and trimm is needed
    if (history.size() > savePathHistoryLength()) {
        history = history.mid(0, savePathHistoryLength());
        historyChanged = true;
    }
    // Save history
    if (historyChanged)
        settings()->storeValue(KEY_SAVEPATHHISTORY, history);
```
I think this solve all the issues, which I described or simply remove the `history.pop_back();` and only use `history.mid()`, it doesn't matter, important is to make it work properly.